src/sys/unix/tcp.rs

```diff
@@ -14,11 +15,13 @@ use sys::unix::eventedfd::EventedFd;
 #[derive(Debug)]
 pub struct TcpStream {
     inner: net::TcpStream,
+    selector_id: Cell<Option<usize>>,
```

Comment:
This has the consequence of making the types not `Sync`, which perhaps isn't desired? Concurrently reading/writing with a `TcpStream` may be useful in some cases.